lib/tools/adb-commands.js

```diff
@@ -1025,19 +1025,33 @@ methods.getLogcatLogs = function () {
 };
 
 /**
- * Set the callback for for logcat output event.
+ * Set the callback for the logcat output event.
  *
  * @param {Function} listener - The listener function, which accepts one argument. The argument is
  *                              a log record object with `timestamp`, `level` and `message` properties.
  * @throws {Error} If logcat process is not running.
  */
 methods.setLogcatListener = function (listener) {
   if (_.isEmpty(this.logcat)) {
-    throw new Error("Can't get logcat logs since logcat hasn't started");
+    throw new Error("Logcat process hasn't been started");
   }
   this.logcat.on('output', listener);
 };
 
+/**
+ * Removes the previously set callback for the logcat output event.
+ *
+ * @param {Function} listener - The listener function, which has been previously
+ *                              passed to `setLogcatListener`
+ * @throws {Error} If logcat process is not running.
+ */
+methods.removeLogcatListener = function (listener) {
```

Comment:
looks good. i often also find it useful to use the [`removeAllListeners`](https://nodejs.org/api/events.html#events_emitter_removealllisteners_eventname) method which i think is available on EventEmitters. sometimes you don't have access to the original listener and don't mind removing all of them.

Reply:
yep, I know about it. Although here you can only assign a listener by using `setLogcatListener` method, so the client knows about the assigned callbacks.